From 6e549ddb51fbebe1778143bc91b8cd434d751870 Mon Sep 17 00:00:00 2001
From: Antoine Martin <dev@ayakael.net>
Date: Wed, 1 Dec 2021 02:34:39 +0000
Subject: [PATCH 1/1] Runtime link order patch for alpine

For some reason, runtime does not link in the right order. This fixes that.
---
 src/installer/corehost/cli/apphost/static/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/installer/corehost/cli/apphost/static/CMakeLists.txt b/src/installer/corehost/cli/apphost/static/CMakeLists.txt
index ffb4f3ed02c..b05a05e70e2 100644
--- a/src/installer/corehost/cli/apphost/static/CMakeLists.txt
+++ b/src/installer/corehost/cli/apphost/static/CMakeLists.txt
@@ -207,12 +207,12 @@ target_link_libraries(
     libhostcommon
     ${CORECLR_LIBRARIES}
 
-    ${ZLIB_LIBRARIES}
-    ${LIBGSS}
-    ${NATIVE_LIBS_EXTRA}
-
     ${START_WHOLE_ARCHIVE}
     ${RUNTIMEINFO_LIB}
     ${NATIVE_LIBS}
     ${END_WHOLE_ARCHIVE}
+
+    ${ZLIB_LIBRARIES}
+    ${LIBGSS}
+    ${NATIVE_LIBS_EXTRA}
 )
-- 
2.34.0

